### PR TITLE
Get DB description and suggested questions

### PR DESCRIPTION
**Pull request by poornimaramesh**
Reviewer:
Primary reviewer: @suzinyou
Secondary reviewer: @ziakhan04

Estimate: ~15 mins

---

### Goal
Add functionality for getting a user-parsable description and suggested questions

### Changes
- Added a new module inside `query_processor` called `db_descriptor`
- Used a single class called DBDescriptor to get and cache descriptions + suggested questions for all databases (hard coded prompts / steps after checking how the crewai agents do this)
- Added a cell in the demo notebook for this

## How has this been tested?
Run the demo notebook (I tested with the TN COVID, ESA GBV and Morocco Afrobarometer DBs)

## TODOS:
- I tested functionality with and without chat history -- doesn't seem liek chat history makes a huge difference on the test DBs for now, so I took it out. We can put it back in after we figure out how we're going to do logging

## To-do before merge
- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated the requirements (if applicable)
- [x] I have updated affected documentation (if applicable)

### CODE REVIEW

File: askametric/query_processor/db_descriptor/descriptor_prompts.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    tables_description: str,
+    db_schema: str,
+    column_description: str = "",
+) -> tuple[str, str]:
+    """
+    Create prompt to get DB description
+    """
+    system = """
+    You are an expert in semantics and contextualization
+    Provide a DESCRIPTION the information available in the database,
```
Comment (suzinyou) [inline askametric/query_processor/db_descriptor/descriptor_prompts.py:12]:
```suggestion
    Provide a DESCRIPTION of the information available in the database,
```

File: askametric/query_processor/db_descriptor/descriptor_prompts.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    system_prompt: str,
+    tables_description: str,
+    db_schema: str,
+    column_description: str = "",
+) -> tuple[str, str]:
+    """
+    Create prompt to get DB description
+    """
+    system = """
+    You are an expert in semantics and contextualization
```
Comment (suzinyou) [inline askametric/query_processor/db_descriptor/descriptor_prompts.py:11]:
```suggestion
    You are an expert in semantics and contextualization.
```

File: askametric/query_processor/db_descriptor/descriptor_prompts.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    Provide a DESCRIPTION the information available in the database,
+    do NOT provide specific numbers.
+    Be clear and descriptive, but concise.
```
Comment (suzinyou) [inline askametric/query_processor/db_descriptor/descriptor_prompts.py:14]: Do you think explicitly giving word limit / range would work better maybe?

**Review by suzinyou — APPROVED** [review]
Thank you Poornima!

The `db_descriptor` module seems independent of query_processor module and maybe can be pulled out of that module? But I might shift things around once I finally get to table and column description generation so totally ok leaving it where it is!

**Review by poornimaramesh — COMMENTED** [review]
Will leave db descriptor where it is for now (you're right -- it's independent). But I imagine it will have interdependencies with the core pipeline at some point, and we can also move around later.

**Pull request merged**